podman/api/client.py

```diff
@@ -24,6 +24,44 @@
 """Type alias for request timeout parameter."""
 
 
+class APIResponse:
+    """APIResponse proxy requests.Response objects.
+
+    Override raise_for_status() to implement Podman API binding errors.
+    All other methods and attributes forwarded to original Response.
+    """
+
+    def __init__(self, response: requests.Response):
+        """Initialize APIResponse.
+
+        Args:
+            response: the requests.Response to provide implementation
+        """
+        self._response = response
+
+    def __getattr__(self, item: str):
+        """Forward any query for an attribute not defined in this proxy class to wrapped class."""
+        return getattr(self._response, item)
+
+    def raise_for_status(
+        self, not_found: Type[APIError] = NotFound
+    ) -> None:  # pylint: disable=arguments-differ
+        """Raises exception when Podman service reports one."""
+        if self.status_code < 400:
```

Comment:
< 300?  What happens if we get a redirect?

Reply:
The requests library takes care of redirects.